server/controllers/authentication.js

```diff
@@ -0,0 +1,35 @@
+import User from '../models/userModel';
+import { buildResponse } from '../utils/helpers';
+
+exports.signUp = (req, res, next) => {
+  const { name, email, password } = req.body;
+  if (!email || !password) {
+    const errorMessage = 'You must provide email and password';
+    const response = buildResponse(false, errorMessage);
+    return res.status(400).send(response);
+  }
+  User.findOne({ email }, (err, existingUser) => {
+    if (err) {
+      return next(err);
+    }
+    if (existingUser) {
+      const errorMessage = 'Email already in use';
+      const response = buildResponse(false, errorMessage);
+      return res.status(400).send(response);
+    }
+    const user = new User({
+      name,
+      email,
+      password,
+    });
+    user.save((error) => {
+      if (error) {
+        return next(error);
+      }
+      const response = buildResponse(true, []);
+      return res.status(200).send(response);
+    });
+    return null;
```

Comment:
Please return with some response.

Reply:
I have added it because of eslint errors. Arrow functions expect to return something at last while I was returning from if statements. I will try to refactor it.